src/unix/mod.rs

```diff
@@ -135,7 +135,7 @@ cfg_if! {
     if #[cfg(not(stdbuild))] {
         // cargo build, don't pull in anything extra as the libstd  dep
         // already pulls in all libs.
-    } else if #[cfg(target_env = "musl")] {
+    } else if #[cfg(all(target_env = "musl", target_arch = "x86_64"))] {
```

Comment:
Hm interesting, that's an unusual difference between the two musl targets. In general linkage to libc is somewhat up-in-the-air (e.g. how it happens), but I guess musl+mips is small enough for now that this should be ok.